### PR TITLE
feat: create color design tokens

### PR DESCRIPTION
**Pull request by nicholasrice**
# Pull Request

## 📖 Description
This PR integrates new color recipes into DesignTokens to be used in CSS. It replaces all instances of color behaviors in components in `@microsoft/fast-components`

### General

- [ ] I have included a change request file using `$ yarn change`
- [ ] I have added tests for my changes.
- [ ] I have tested my changes.
- [ ] I have updated the project documentation to reflect my changes.
- [ ] I have read the [CONTRIBUTING](https://github.com/Microsoft/fast/blob/master/CONTRIBUTING.md) documentation and followed the [standards](https://www.fast.design/docs/community/code-of-conduct/#our-standards) for this project.

### Component-specific

- [ ] I have added a new component
- [ ] I have modified an existing component
- [ ] I have updated the [definition file](https://github.com/Microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#definition)
- [ ] I have updated the [configuration file](https://github.com/Microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#configuration)

### CODE REVIEW

**Comment by EisenbergEffect** [comment]
How was the perf looking in the end? Any additionally follow-up work planned for that?

**Comment by nicholasrice** [comment]
@EisenbergEffect the color recipes themselves seem fine for perf, but the frequent resolution of token values (colors often depend on lots of tokens, and each element often depends on lots of colors) is causing a significant perf regression. The problem grows linearly with the number of elements. I'm going to hold off on merging this until I can do a deeper investigation and find a resolution.

**Comment by EisenbergEffect** [comment]
Could we store the token values themselves in a central storage service, but have it be something that is created on the fly as needed? For example, set up a singleton storage on the element where the design system is initialized and have everything reference that. But, if someone needs to set a token value somewhere on a child node, we dynamically create a new storage service on that node and switch the system over from singleton mode to dom hierarchy mode and then have consumers locate the storage system in the same way that it located the design system provider in the past.

**Comment by nicholasrice** [comment]
We *sort of* do that already but there might be some short-circuiting we can do during resolution. The core use-case is for tokens to be set *frequently* throughout the DOM hierarchy (think fill/background color & palettes) so in practice the system will enter the more inefficient mode rather quickly. I do have it set up where each token is isolated though, so we could see some performance gains if we only branch the overridden token to that mode.

**Comment by EisenbergEffect** [comment]
I can't remember the internal implementations details, so forgive me if this is rehashing or we've already tried this. But, what if we only acted on "write" of a token and never on "read". The idea would be that the token simply injects the CSS variable and nothing else. Then, the design system writes all the token values, performing all calculations. We do this as part of the initial registration. Then, if someone writes a new token value to a DOM node, we publish an event up the tree to find the parent node where tokens were written. We then pull down all the tokens and write them all to new the node where the new value is. That would cause all css variables to trickle down. Would something like this work? My thought is that the writing of custom properties is less frequent than the need to access them in styles. And then maybe we establish some practices/patterns related to this. So, for example, with cards, we encourage folks to override the token on the parent node that contains the list of cards, rather than on individual cards, so we get less frequent needs to write all tokens. We might also be able to reduce the number of events to find token storage by doing a fast path to first check the parentNode before firing an event. Just tossing ideas out...

File: packages/web-components/fast-components/src/accordion-item/accordion-item.styles.ts [hunk 1]
```diff
@@ -58,17 +52,17 @@ export const accordionItemStyles = (context, definition) =>
...
 
     .button:hover {
-        color: ${neutralForegroundHoverBehavior.var};
+        color: ${neutralForegroundRest};
     }
 
     .button:active {
-        color: ${neutralForegroundActiveBehavior.var};
+        color: ${neutralForegroundRest};
     }
```
Comment (chrisdholt) [inline packages/web-components/fast-components/src/accordion-item/accordion-item.styles.ts:66]: @nicholasrice should these all be neutralForegroundRest or should we also have Hover and Active here?

Comment (nicholasrice) [reply packages/web-components/fast-components/src/accordion-item/accordion-item.styles.ts:66]: In practice `neutralForeground` never actually had hover/active states. They were all the same, so I just made a single neutralForegroundRest recipe.

Comment (chrisdholt) [reply packages/web-components/fast-components/src/accordion-item/accordion-item.styles.ts:66]: makes sense!

**Review by chrisdholt — APPROVED** [review]

**Comment by nicholasrice** [comment]
I ended up adding an optional argument for Fill recipes to provide a the swatch context to calculate from. If no swatch is provided, the fillColor is retrieved from the target element. This is ready for review now; In a follow up PR I would go through and delete old assets and update documentation.

**Review by EisenbergEffect — APPROVED** [review]

**Pull request merged**